test/gtest/uct/ib/test_ib.cc

```diff
@@ -439,26 +439,66 @@ class test_uct_ib_sl : public test_uct_ib_with_specific_port {
         }
     }
 
+    bool transport_has_ddp()
+    {
+        ucs::handle<uct_md_h> uct_md;
+
+        if (!has_transport("rc_mlx5") && !has_transport("dc_mlx5")) {
+            return false;
+        }
+
+        UCS_TEST_CREATE_HANDLE(uct_md_h, uct_md, uct_md_close, uct_md_open,
+                               &uct_ib_component,
+                               ibv_get_device_name(m_ibctx->device),
+                               m_md_config);
+
+        uct_ib_mlx5_md_t *ib_md = ucs_derived_of(uct_md, uct_ib_mlx5_md_t);
+        int rc_has_ddp          = has_transport("rc_mlx5") &&
+                                  (ib_md->dp_ordering_cap.rc ==
+                                   UCT_IB_MLX5_DP_ORDERING_OOO_ALL);
+        int dc_has_ddp          = has_transport("dc_mlx5") &&
+                                  (ib_md->dp_ordering_cap.dc ==
+                                   UCT_IB_MLX5_DP_ORDERING_OOO_ALL);
+        return rc_has_ddp || dc_has_ddp;
+    }
+
+    void set_ddp_enable(bool enable)
+    {
+        if (has_transport("rc_mlx5")) {
+            modify_config("RC_MLX5_DDP_ENABLE", enable ? "y" : "n");
+        }
+
+        if (has_transport("dc_mlx5")) {
+            modify_config("DC_MLX5_DDP_ENABLE", enable ? "y" : "n");
+        }
+    }
+
 protected:
     uint16_t m_ooo_sl_mask;
 };
 
 UCS_TEST_P(test_uct_ib_sl, check_ib_sl_config) {
+    bool has_ddp = transport_has_ddp();
+
     // go over all SLs, check UCTs could be initialized on a specific SL
     // and able to send/recv traffic
-    for (uint8_t sl = 0; sl < UCT_IB_SL_NUM; ++sl)  {
+    for (uint8_t sl = 0; sl < UCT_IB_SL_NUM; ++sl) {
+        uint16_t sl_supports_ar = (m_ooo_sl_mask & UCS_BIT(sl));
         if (!has_transport("rc_verbs") && !has_transport("ud_verbs")) {
             // if AR is configured on the given SL, set AR_ENABLE to "y",
             // otherwise - to "n" in order to test that AR_ENABLE parameter
             // works as expected w/o errors and warnings
-            modify_config("IB_AR_ENABLE",
-                          (m_ooo_sl_mask & UCS_BIT(sl)) ? "y" : "n");
+            modify_config("IB_AR_ENABLE", sl_supports_ar ? "y" : "n");
         }
+
         modify_config("IB_SL", ucs::to_string(static_cast<uint16_t>(sl)));
 
-        test_uct_ib::init();
-        send_recv_short();
-        test_uct_ib::cleanup();
+        for (int test_ddp = 0; test_ddp <= has_ddp; ++test_ddp) {
+            set_ddp_enable(test_ddp && sl_supports_ar);
```

Comment:
break if not supported?

Reply:
Why?
The goal here is to do 2 iterations when ddp is supported: one with ddp and another one without ddp,
when ddp is not supported we want to do a single iteration - without ddp, just to test AR / different SLs

Reply:
from what I understand when `has_dpp=1` and `sl_supports_ar=0`:
- test_ddp=0: set_ddp_enable(0); init(); recv_short(); cleanup();
- test_dpp=1: set_ddp_enable(1 && 0): recv_short(); cleanup(); 

so I think we do the same sequence twice. maybe if we use `for () {}` then check ` <= (has_ddp && sl_supports_ar)` and `set_ddp_enable(test_ddp)`?